### PR TITLE
#VP-2385 job settings on the fly

### PR DESCRIPTION
**Pull request by kostyrin**

### CODE REVIEW

File: src/VirtoCommerce.Platform.Core/Jobs/RecurringJobExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            }
+        }
+
+
+        private static async Task RunOrRemoveJobAsync(this IRecurringJobManager recurringJobManager, ISettingsManager settingsManager, SettingCronJob settingCronJob)
+        {
+            var processJobEnable = await settingsManager.GetValueAsync(settingCronJob.EnableSetting.Name, (bool)settingCronJob.EnableSetting.DefaultValue);
+            if (processJobEnable)
+            {
+                recurringJobManager.RemoveIfExists(settingCronJob.RecurringJobId);
```
Comment (yecli) [inline src/VirtoCommerce.Platform.Core/Jobs/RecurringJobExtensions.cs:106]: Seems `RemoveIfExists` could be done before the `if` statement, so `else` would not be needed at all.

Comment (kostyrin) [reply src/VirtoCommerce.Platform.Core/Jobs/RecurringJobExtensions.cs]: remove

File: src/VirtoCommerce.Platform.Hangfire/Extensions/RecurringJobExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+            var settingCronJob = new SettingCronJobBuilder(new SettingCronJob())
+                .SetEnablerSetting(enablerSetting)
+                .SetCronSetting(cronSetting)
+                .SetJobId(jobId)
+                .SetQueueName(queue)
+                .SetTimeZoneInfo(timeZoneInfo)
+                .ToJob(methodCall)
+                .Build();
+
+            _observedSettingsDict.AddOrUpdate(settingCronJob.EnableSetting.Name, settingCronJob, (settingName, сronJob) => settingCronJob);
```
Comment (tatarincev) [inline src/VirtoCommerce.Platform.Hangfire/Extensions/RecurringJobExtensions.cs:87]: Duplicate code you might call exists overridden method `WatchJobSettingAsync` instead

File: src/VirtoCommerce.Platform.Hangfire/Extensions/RecurringJobExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        /// </summary>
+        /// <param name="recurringJobManager"></param>
+        /// <param name="settingsManager"></param>
+        /// <param name="settingCronJob"></param>
+        /// <returns></returns>
+        public static Task WatchJobSettingAsync(this IRecurringJobManager recurringJobManager,
+            ISettingsManager settingsManager,
+            SettingCronJob settingCronJob)
+        {
+            if (recurringJobManager == null) throw new ArgumentNullException(nameof(recurringJobManager));
```
Comment (tatarincev) [inline src/VirtoCommerce.Platform.Hangfire/Extensions/RecurringJobExtensions.cs:46]: use please curly brackets instead please

File: src/VirtoCommerce.Platform.Hangfire/Models/SettingCronJob.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
+using System;
+using Hangfire.States;
+using Hangfire.Common;
+using VirtoCommerce.Platform.Core.Settings;
+
+namespace VirtoCommerce.Platform.Hangfire
+{
+    public class SettingCronJob
```
Comment (tatarincev) [inline src/VirtoCommerce.Platform.Hangfire/Models/SettingCronJob.cs:8]: can we mark this class as internal?

Comment (kostyrin) [reply src/VirtoCommerce.Platform.Hangfire/Models/SettingCronJob.cs]: can't. only mark ctor as internal. is that enough?

**Pull request merged**